airbyte-integrations/connectors/source-snowflake/source_snowflake/streams.py

```diff
@@ -0,0 +1,566 @@
+            SHOW TABLES IN DATABASE statement does not include "system tables" in the dataset0
+            if schema is provided by the use we replace the search of tables (streams) in database by search in shema
+
+        TODO: Validate that the streams in the pushdown filter configuration are available
+        """
+        database = self.config["database"]
+        schema = self.config.get('schema', "")
+        if not schema:
+            return f"SHOW TABLES IN DATABASE {database}"
+
+        return f"SHOW TABLES IN SCHEMA {database}.{schema}"
+
+    def request_body_json(
+            self,
+            stream_state: Optional[Mapping[str, Any]],
+            stream_slice: Optional[Mapping[str, Any]] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping[str, Any]]:
+        json_payload = {
+            "statement": self.statement,
+            "role": self.config['role'],
+            "warehouse": self.config['warehouse'],
+            "database": self.config['database'],
+            "timeout": "1000",
+        }
+        schema = self.config.get('schema', '')
+        if schema:
+            json_payload['schema'] = schema
+        return json_payload
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
+        """
+        response_json = response.json()
+        records = response_json.get("data", [])
+        yield from records
+
+
+class TableCatalogStream(SnowflakeStream):
+    DATABASE_NAME_COLUMN = "name"
+    SCHEMA_NAME_COLUMN = "schema_name"
+
+    def __init__(self, url_base, config, **kwargs):
+        super().__init__(**kwargs)
+        self._url_base = url_base
+        self.config = config
+
+    def path(
+            self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        """
+            path of request
+        """
+
+        return f"{self.url_base}/{self.url_suffix}"
+
+    @property
+    def url_base(self):
+        return self._url_base
+
+    @property
+    def statement(self):
+        database = self.config["database"]
+        schema = self.config.get('schema', '')
+        if not schema:
+            return f"SHOW TABLES IN DATABASE {database}"
+
+        return f"SHOW TABLES IN SCHEMA {database}.{schema}"
+
+    def request_body_json(
+            self,
+            stream_state: Optional[Mapping[str, Any]],
+            stream_slice: Optional[Mapping[str, Any]] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping[str, Any]]:
+        json_payload = {
+            "statement": self.statement,
+            "role": self.config['role'],
+            "warehouse": self.config['warehouse'],
+            "database": self.config['database'],
+            "timeout": "1000",
```

Comment:
This is the behavior of timeout following the [documentation](https://docs.snowflake.com/en/developer-guide/sql-api/reference):
timeout
(Optional) Timeout in seconds for statement execution. If the execution of a statement takes longer than the specified timeout, the execution is automatically canceled. To set the timeout to the maximum value (604800 seconds), set timeout to 0. If this field is not set, the timeout specified by the [STATEMENT_TIMEOUT_IN_SECONDS](https://docs.snowflake.com/en/sql-reference/parameters.html#label-statement-timeout-in-seconds) parameter is used.
 I put this timeout by default, we can change it if you want.

Reply:
So the API default timeout it 16 minutes and the max timeout is 7 days? Those are pretty generous timeouts haha!

I don't know enough about how long our requests will be. I would assume though that the user impact is lessen if the timeout is too big than too small. I think that 16 minutes (i.e. 1000 seconds) should be fine. I would maybe extract this as a variable so that all the HTTP requests use the same and document next to this constant that we are using 1000 seconds without much basis for now and this value can be updated if we get more information